sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3129,6 +3129,28 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
                   ZeDevicePciProperties.address.function);
     return ReturnValue(AddressBuffer);
   }
+
+  case PI_EXT_INTEL_DEVICE_INFO_FREE_MEMORY: {
+    if (getenv("ZES_ENABLE_SYSMAN") == nullptr) {
+      setErrorMessage("Set ZES_ENABLE_SYSMAN=1 to obtain free memory",
+                      PI_SUCCESS);
+      return PI_ERROR_PLUGIN_SPECIFIC_ERROR;
+    }
+    uint64_t FreeMemory = 0;
+    uint32_t MemCount = 0;
+    ZE_CALL(zesDeviceEnumMemoryModules, (ZeDevice, &MemCount, nullptr));
+    std::vector<zes_mem_handle_t> MemHandles(MemCount);
+    ZE_CALL(zesDeviceEnumMemoryModules,
+            (ZeDevice, &MemCount, MemHandles.data()));
+
+    for (auto &ZesMemHandle : MemHandles) {
+      ZesStruct<zes_mem_state_t> ZeMemState;
+      ZE_CALL(zesMemoryGetState, (ZesMemHandle, &ZeMemState));
+      FreeMemory += ZeMemState.free;
+    }
```

Comment:
Should we take into account only specific type of memory here?
For example, the list of memory types includes ZES_MEM_TYPE_SLM and ZES_MEM_TYPE_GRF: https://spec.oneapi.io/level-zero/latest/sysman/api.html?highlight=zesdeviceenummemorymodules#memory-enums
In other words, is it correct to treat all types of memory as  the global device memory in this case?

Reply:
I've added a check HBM type as that is where zeMemAllocDevice allocates from.

Reply:
Thank you!